### ISSUE TITLE
Add tree-sitter / web-tree-sitter

### ISSUE THREAD

**Issue by 0xdevalias** [sxzz/ast-explorer#25]
### Clear and concise description of the problem

As a developer using `ast-explorer`, I want to be able to see the AST's parsed by `tree-sitter` / `web-tree-sitter`

### Suggested solution

It would be really cool if AST Explorer was able to support 'Semantic' and/or 'Tree-Sitter':

- https://github.com/github/semantic
  - > Parsing, analyzing, and comparing source code across many languages
- https://github.com/tree-sitter/tree-sitter
  - > An incremental parsing system for programming tools
  - https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/
    - > WebAssembly bindings to the Tree-sitter parsing library.
  - https://github.com/tree-sitter/tree-sitter/tree/master/cli
    - > The Tree-sitter CLI allows you to develop, test, and use Tree-sitter grammars from the command line.
- https://github.com/tree-sitter/node-tree-sitter
  - > Node.js bindings for tree-sitter

---

> Tree sitter's WASM distribution seems to be able to create an AST in browser:
> 
> [npmjs.com/package/web-tree-sitter](https://www.npmjs.com/package/web-tree-sitter)
> 
> _Originally posted by @privatenumber in https://github.com/fkling/astexplorer/issues/622#issue-1087317072_

See also:

- https://github.com/tree-sitter/tree-sitter/tree/master/lib/binding_web

### Additional context

- https://github.com/fkling/astexplorer/issues/688
- https://github.com/fkling/astexplorer/issues/622

### Validations

- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sxzz** [comment]
I tried `web-tree-sitter`, but it seems like it cannot be loaded via CDN. There's a path error, which could be caused by `tree-sitter` itself, and it's even not an ESM module...

**Comment by LoTwT** [comment]
Here is a [demo](https://stackblitz.com/edit/vitejs-vite-skm25yrv?file=src%2Fcomponents%2Ftest.vue) that uses `web-tree-sitter` and `tree-sitter-javascript` to parse JavaScript code in the browser. They are all dynamically imported. Could it be used?

**Comment by sxzz** [comment]
Certainly, it can serve as basic support. However, since it’s not JSON, it cannot be treated as a tree structure.

**Comment by LoTwT** [comment]
`tree-sitter` doesn't define an official JSON structure. How about defining the JSON structure ourselves, just like `ast-grep` does?

The node type in `ast-grep` is [DumpNode](https://github.com/ast-grep/ast-grep.github.io/blob/ecb762fa71bc18df34c79fa53dbb96fa30fcc30e/website/src/components/dump/dumpTree.ts#L23). The method is `dumpASTNodes` powered by `ast-grep-wasm` which hasn't been published yet.

**Comment by sxzz** [comment]
Sure
